File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-213846-32199/analysis/own-tests-project/src/main/java/customerimporter/Identification.java
package customerimporter;

import java.util.List;
import java.util.function.Function;
import java.util.function.Supplier;

public abstract class Identification {

    private static final List<IdentificationType> types = List.of(
        new IdentificationType(DNI.TYPE, DNI::from),
        new IdentificationType(CUIT.TYPE, CUIT::from));

    // instance creation

    public static Identification forTypeAndNumber(String aType, String aNumber) {
        return types.stream()
            .filter(anIdentificationType -> anIdentificationType.handles(aType))
            .findFirst()
            .orElseThrow(() -> new RuntimeException(invalidIdentificationTypeErrorDescription()))
            .createFrom(aNumber);
    }

    // error messages

    public static String invalidIdentificationTypeErrorDescription() {
        return "Invalid identification type";
    }

    // identification

    public abstract String number();

    public abstract String type();

    public boolean isFor(String aType, String aNumber) {
        return type().equals(aType) && number().equals(aNumber);
    }

    // testing

    public abstract boolean isCUIT();

    public abstract boolean isDNI();

    // numbers

    public abstract Object cuitNumberIfNone(Supplier<Object> aNoneBlock);

    public abstract Object dniNumberIfNone(Supplier<Object> aNoneBlock);

    // types

    private record IdentificationType(String type, Function<String, Identification> creator) {

        public boolean handles(String aType) {
            return type.equals(aType);
        }

        public Identification createFrom(String aNumber) {
            return creator.apply(aNumber);
        }
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-202259-53403/analysis/acceptance-project/src/customerimporter/identification.py
from abc import abstractmethod

from customerimporter.imported_value import ImportedValue


class Identification(ImportedValue):

    # instance creation

    @classmethod
    def for_type_and_number(cls, an_identification_type, an_identification_number):
        return cls.class_handling(an_identification_type).with_value(an_identification_number)

    # type

    @classmethod
    def can_handle(cls, an_identification_type):
        return an_identification_type == cls.type_code()

    @classmethod
    @abstractmethod
    def type_code(cls):
        pass

    def identification_type(self):
        return type(self).type_code()

    # testing

    def is_cuit(self):
        return False

    def is_dni(self):
        return False

    def is_of_type_and_number(self, an_identification_type, an_identification_number):
        return self.identification_type() == an_identification_type and self.number() == an_identification_number

    # number

    def cuit_number_if_none(self, a_none_closure):
        return a_none_closure()

    def dni_number_if_none(self, a_none_block):
        return a_none_block()

    def number(self):
        return self.value()
